pkg/autodiscovery/listeners/dbm_aurora.go

```diff
@@ -0,0 +1,272 @@
+	auroraPostgresqlEngine: dbmPostgresADIdentifier,
+	auroraMysqlEngine:      dbmMySQLADIdentifier,
+}
+
+var _ Service = &DBMAuroraService{}
+
+// DBMAuroraService implements and store results from the Service interface for the DBMAuroraListener
+type DBMAuroraService struct {
+	adIdentifier string
+	entityID     string
+	checkName    string
+	clusterID    string
+	region       string
+	instance     *aws.Instance
+}
+
+// NewDBMAuroraListener returns a new DBMAuroraListener
+func NewDBMAuroraListener(Config) (ServiceListener, error) {
+	config, err := dbmconfig.NewAuroraAutodiscoveryConfig()
+	if err != nil {
+		return nil, err
+	}
+	client, region, err := aws.NewRDSClient()
+	if err != nil {
+		return nil, err
+	}
+	config.Region = region
+	return newDBMAuroraListener(config, client, nil), nil
+}
+
+func newDBMAuroraListener(config dbmconfig.AuroraConfig, awsClient aws.RDSClient, ticks <-chan time.Time) ServiceListener {
+	l := &DBMAuroraListener{
+		config:       config,
+		services:     make(map[string]Service),
+		stop:         make(chan bool),
+		awsRdsClient: awsClient,
+		ticks:        ticks,
+	}
+	if l.ticks == nil {
+		l.ticker = time.NewTicker(time.Duration(l.config.DiscoveryInterval) * time.Second)
+		l.ticks = l.ticker.C
+	}
+	return l
+}
+
+// Listen listens for new and deleted aurora endpoints
+func (l *DBMAuroraListener) Listen(newSvc, delSvc chan<- Service) {
+	l.newService = newSvc
+	l.delService = delSvc
+	go l.run()
+}
+
+// Stop stops the listener
+func (l *DBMAuroraListener) Stop() {
+	l.stop <- true
+	if l.ticker != nil {
+		l.ticker.Stop()
+	}
+}
+
+// run is the main loop for the aurora listener discovery
+func (l *DBMAuroraListener) run() {
+	for {
+		l.discoverAuroraClusters()
+		select {
+		case <-l.stop:
+			return
+		case <-l.ticks:
+		}
+	}
+}
+
+// discoverAuroraClusters discovers aurora clusters according to the configuration
+func (l *DBMAuroraListener) discoverAuroraClusters() {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(l.config.QueryTimeout)*time.Second)
+	defer cancel()
+	ids, err := l.awsRdsClient.GetAuroraClustersFromTags(ctx, l.config.Tags)
+	if err != nil {
+		_ = log.Error(err)
+		return
+	}
+	if len(ids) == 0 {
+		log.Debugf("no aurora clusters found with provided tags %v", l.config.Tags)
+		return
+	}
+	auroraCluster, err := l.awsRdsClient.GetAuroraClusterEndpoints(ctx, ids)
+	if err != nil {
+		_ = log.Error(err)
+		return
+	}
+	discoveredServices := make(map[string]struct{})
+	for id, c := range auroraCluster {
+		for _, instance := range c.Instances {
+			if instance == nil {
+				_ = log.Warnf("received malformed instance response for cluster %s, skipping", id)
+				continue
+			}
+			entityID := instance.Digest(engineToIntegrationType[instance.Engine], id)
+			discoveredServices[entityID] = struct{}{}
+			l.createService(entityID, id, instance)
+		}
+	}
+	// TODO: should we wait a certain number of run iterations before we remove instances?
```

Comment:
Is this TODO still relevant? If so we may want to add more clarity and what needs to be done to resolve it.

Reply:
I don't think so, if it becomes an issue we can do something about this later on. I think i'll likely add caching of the instances as a follow-up at some point. This is going to be a beta feature for customers, so we can collect feedback on it and go from there. I'll remove the TODO in the meantime